### PR TITLE
New pull request for sci-libs/vxl

### PR DESCRIPTION
**Pull request by TheChymera**
Dependency for [Freesurfer](https://github.com/gentoo-science/sci/issues/401).

### CODE REVIEW

File: sci-libs/vxl/ChangeLog [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# ChangeLog for sci-libs/vxl
+# Copyright 1999-2015 Gentoo Foundation; Distributed under the GPL v2
+# $Header: $
+
+*vxl-9999 (25 May 2015)
+
+  25 May 2015;  <chymera@gentoo.org> +metadata.xml, +vxl-9999.ebuild:
```
Comment (jlec) [inline sci-libs/vxl/ChangeLog]: really? ;)

File: sci-libs/vxl/vxl-9999.ebuild [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+HOMEPAGE="http://vxl.sourceforge.net"
+SRC_URI=""
+EGIT_REPO_URI="http://git.code.sf.net/p/vxl/git"
+
+LICENSE="BSD"
+SLOT="0"
+IUSE=""
+KEYWORDS=""
+
+RDEPEND=""
```
Comment (jlec) [inline sci-libs/vxl/vxl-9999.ebuild]: empty, so please drop.

File: sci-libs/vxl/vxl-9999.ebuild [hunk 3]
```diff
@@ -0,0 +1,20 @@
+# Copyright 1999-2015 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Header: $
+
+EAPI="5"
+
+inherit git-r3 cmake-utils
```
Comment (jlec) [inline sci-libs/vxl/vxl-9999.ebuild]: please sort alphabetically.

File: sci-libs/vxl/ChangeLog (hunk 1 above)
Comment (TheChymera) [inline sci-libs/vxl/ChangeLog]: you mean my email? yes, seriously, I have no idea where this is coming from and how to fix it. You probably remember that I tried to (with help from here and from the IRC) repeatedly :-/

**Comment by kiwifb** [comment]
So you mean that your variable `ECHANGELOG_USER` is set to the right thing and exported and you still get a default?

**Comment by TheChymera** [comment]
@kiwifb 

```
chymera@zenbookhost ~ $ echo $ECHANGELOG_USER
Horea Christian <h.chr@mail.ru>
```

**Comment by kiwifb** [comment]
And you have `gentoolkit-dev` installed?

**Comment by TheChymera** [comment]
@kiwifb ah, no, just plain gentoolkit

**Comment by TheChymera** [comment]
@kiwifb ok, got gentoolkit-dev, what now? should I remove plain gentoolkit?

**Comment by kiwifb** [comment]
Nope they are complimentary of each other. `gentoolkit` is tools for normal users, `gentoolkit-dev` is extra tools for developers. You get 4 new tools including `echangelog` which is the application normally using the above variable. I am thinking you got a default setting from `repoman` because the call to `echangelog` failed.

**Comment by marbre** [comment]
That shoud not be the problem. I haven't gentoolkit-dev installed and repoman adds the correct e-mail to the ChangeLog.

**Comment by TheChymera** [comment]
for the time being, can I just correct the email and commit with git?

**Comment by kiwifb** [comment]
Not my decision to make but I realised that repoman probably doesn't use `ECHANGELOG_USER` but it would the content of `.gitconfig` mine is looking like
```
[user]
        name = François Bissey
        email = francois.bissey@canterbury.ac.nz
        signingkey = XXXXXXXXXX
```

**Comment by TheChymera** [comment]
@kiwifb 

my `.gitconfig` reads:

```
[user]
	email = h.chr@mail.ru
	name = Horea Christian
	signingkey = XXXXXXXXX
[credential]
	helper = cache --timeout=99999
[push]
	default = simple
```

What does the signingkey do? In addition to repoman inventing email addresses for me, it also does me the courtesy of always asking for my PGP (or whatever) key password every time I commit. Any idea if I can make it stop (e.g. by modifying what signingkey say?)?

**Comment by kiwifb** [comment]
Signing is enabled at the overlay level in `metadata/layout.conf` do not mess up with that. This is the part of `repoman` that gets the committer information
```
def get_committer_name(env=None):
	"""Generate a committer string like echangelog does."""
	if env is None:
		env = os.environ
	if 'GENTOO_COMMITTER_NAME' in env and \
		'GENTOO_COMMITTER_EMAIL' in env:
		user = '%s <%s>' % (env['GENTOO_COMMITTER_NAME'],
			env['GENTOO_COMMITTER_EMAIL'])
	elif 'GENTOO_AUTHOR_NAME' in env and \
			'GENTOO_AUTHOR_EMAIL' in env:
		user = '%s <%s>' % (env['GENTOO_AUTHOR_NAME'],
			env['GENTOO_AUTHOR_EMAIL'])
	elif 'ECHANGELOG_USER' in env:
		user = env['ECHANGELOG_USER']
	else:
		pwd_struct = pwd.getpwuid(os.getuid())
		gecos = pwd_struct.pw_gecos.split(',')[0]  # bug #80011
		user = '%s <%s@gentoo.org>' % (gecos, pwd_struct.pw_name)
	return user
```
so `ECHANGELOG_USER` should be the right thing. From a python shell try:
```
import os
env=os.environ
env['ECHANGELOG_USER']
```

**Comment by TheChymera** [comment]
@kiwifb 

maybe this is the issue?

```
Python 2.7.9 (default, Mar 13 2015, 03:46:20) 
[GCC 4.8.3] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import os
>>> env=os.environ
>>> env["ECHANGELOG_USER"]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib64/python2.7/UserDict.py", line 23, in __getitem__
    raise KeyError(key)
KeyError: 'ECHANGELOG_USER'
>>> env['ECHANGELOG_USER']
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib64/python2.7/UserDict.py", line 23, in __getitem__
    raise KeyError(key)
KeyError: 'ECHANGELOG_USER'
```

**Comment by kiwifb** [comment]
Yes that looks like a problem. And if you start python with `ECHANGELOG_USER=me python`?

**Comment by TheChymera** [comment]
@kiwifb 

That seems to work:

```
chymera@zenbookhost ~/src/neurogentoo/sci-libs/minc $ ECHANGELOG_USER=me python
Python 2.7.9 (default, Mar 13 2015, 03:46:20) 
[GCC 4.8.3] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import os
>>> env=os.environ
>>> env['ECHANGELOG_USER']
'me'
```

In a possibly related matter, recently, repoman stopped working at all:

```
chymera@zenbookhost ~/src/neurogentoo/sci-libs/minc $ repoman commit
!!! Section 'neurogentoo' in repos.conf has name different from repository name 'science' set inside repository

RepoMan scours the neighborhood...
>>> Creating Manifest for /home/chymera/src/neurogentoo/sci-libs/minc
  RDEPEND.suspect               1
   sci-libs/minc/minc-2.2.00.ebuild: 'dev-util/byacc'

Note: use --include-dev (-d) to check dependencies for 'dev' profiles

* 3 files being committed... 
gpg: skipped "522AF697": No secret key
gpg: signing failed: No secret key
error: gpg failed to sign the data
fatal: failed to write commit object
!!! Exiting on git (shell) error code: 128
```

Any ideas?

**Comment by kiwifb** [comment]
Right it means that your `ECHANGELOG_USER` is not properly exported, what shell do you use? For repoman, well I think you try to be to clever with your overlay setup. Your repo is just a fork of the science overlay so `profiles/repo_name` is probably still science but you try to use another name in `repos.conf` and repoman just doesn't like that.

**Comment by TheChymera** [comment]
@kiwifb regarding repoman not working, the `repo_name` is not an issue, it always gives me the warning, but it worked in spite of it for a long time, the problem was the missing key, for some reason gnome deleted my key...

Anyhow, I use the normal shell, I guess it is called shell or bash or something like that? And I use it via the GNOME terminal. Never did anything fancy regarding my shell or python configuration, so I am surprised stuff is not working... How can I further debug this?

**Comment by TheChymera** [comment]
@kiwifb no ideas?

**Comment by kiwifb** [comment]
Not really. On the one hand you seem to have the variable in your environment but is not exported to applications.

Try putting an alias in `~/.bashrc`
```
alias repoman="ECHANGELOG_USER=\"your name and email\" repoman"
```
should be effective in your next shell session, the command `alias` should list it and any other aliases you have defined.

**Comment by kiwifb** [comment]
Something stupid, I presume you defined `ECHANGELOG_USER` in `~/.bashrc`, did you do
```
ECHANGELOG_USER=.....
```
which certainly won't work, or
```
export ECHANGELOG_USER=.....
```
which should work?

**Comment by TheChymera** [comment]
@kiwifb yes, 

missing export in 

```
export ECHANGELOG_USER="Horea Christian <h.chr@mail.ru>"
```

was my issue (I think). Is there an easy way to use (the now pressumably correctly working) repoman to fix previous commit errors?

**Comment by kiwifb** [comment]
Apart from editing the `Changelog`? Not that I know of. You may be able to use `sed`. From the top of the overlay:
```
sed -i "s:$wrong_stuff:$right_stuff:g" `grep -rl $wrong_stuff *`
git add *
git status
```
The last bit will tell which files are ready to commit (with repoman) to register your changes.

**Comment by marbre** [comment]
Merge the latest master into your branch/repository. Otherwise you'll run into merge conflicts after you fixed the `ChangeLog` files, as I already corrected the ones with commit e1609a961c6d3140ba3a53f64665bf2e3a25d910 that were merged into the sci master.

**Comment by TheChymera** [comment]
can we pull this now and I'll correct my emails later in a separate commit?

File: sci-libs/vxl/vxl-9999.ebuild [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+HOMEPAGE="http://vxl.sourceforge.net"
+SRC_URI=""
+EGIT_REPO_URI="http://git.code.sf.net/p/vxl/git"
+
+LICENSE="BSD"
+SLOT="0"
+IUSE=""
+KEYWORDS=""
+
+
```
Comment (jlec) [inline sci-libs/vxl/vxl-9999.ebuild]: This triggers repoman warnings. Please fix.

File: sci-libs/vxl/ChangeLog [hunk 5]
```diff
@@ -0,0 +1,11 @@
+# ChangeLog for sci-libs/vxl
+# Copyright 1999-2015 Gentoo Foundation; Distributed under the GPL v2
+# $Header: $
+
+  25 May 2015;  <chymera@gentoo.org> vxl-9999.ebuild:
+  dropped emty dep, sorted eclasses
+
+*vxl-9999 (25 May 2015)
+
+  25 May 2015;  <chymera@gentoo.org> +metadata.xml, +vxl-9999.ebuild:
```
Comment (jlec) [inline sci-libs/vxl/ChangeLog]: Correct the email addresses here.

**Comment by TheChymera** [comment]
ok, @jlec , done?

**Pull request merged**